src/components/organisms/UiMenu/_internal/UiMenuItem.vue

```diff
@@ -1,97 +1,91 @@
 <template>
+  <!-- :has-suffix, :icon  -->
   <UiListItem
-    ref="menuItem"
-    :list-item-attrs="defaultProps.listItemAttrs"
+    ref="menuItemTemplateRefs"
+    v-bind="filteredAttrs"
     :tag="UiButton"
+    :has-suffix="hasSuffix"
+    :class="menuItemClass"
     :tabindex="tabindex"
-    :class="[
-      'ui-button--outlined ui-menu-item__button', buttonClass,
-    ]"
+    :list-item-attrs="defaultProps.listItemAttrs"
+    :suffix-attrs="defaultProps.suffixAttrs"
+    @blur="handleMenuItemBlur"
+    @focus="handleMenuItemFocus"
   >
-    <!-- @slot Use this slot to replace label template. -->
-    <slot name="label">
-      <!-- @slot Use this slot to replace label template. -->
-      <span class="ui-menu-item__label">
-        <!-- @slot Use this slot to place label content inside menu-item. -->
-        <slot />
-      </span>
-    </slot>
-
-    <template #suffix>
-      <!-- @slot Use this slot to replace suffix template -->
+    <!-- Allow to use UiListItemSlots / BEGIN -->
+    <template
+      v-for="(_, name) in $slots"
+      #[name]="data"
+    >
       <slot
-        name="suffix"
-        v-bind="{
-          hasSuffix,
-          suffixAttrs: defaultProps.suffixAttrs,
-        }"
-      >
-        <UiMenuItemSuffix
-          v-if="hasSuffix"
-          v-bind="defaultProps.suffixAttrs"
-          class="ui-menu-item__suffix"
-        />
-      </slot>
+        v-bind="data"
+        :name="name"
+      />
     </template>
+    <!-- END -->
+    <slot />
   </UiListItem>
 </template>
 
 <script setup lang="ts">
 import {
   ref,
   computed,
-  onMounted,
-  nextTick,
-  inject,
-  type Ref,
+  useAttrs,
 } from 'vue';
+import UiListItem, { type ListItemAttrsProps } from '../../UiList/_internal/UiListItem.vue';
 import UiButton from '../../../atoms/UiButton/UiButton.vue';
-import UiListItem from '../../UiList/_internal/UiListItem.vue';
-import type { ListItemAttrsProps } from '../../UiList/_internal/UiListItem.vue';
-import UiMenuItemSuffix from './UiMenuItemSuffix.vue';
-import { useAttributes } from '../../../../composable';
 import type { DefineAttrsProps } from '../../../../types';
 import type { MenuItemProps } from './MenuItemProps';
-import type { MenuItem } from '../UiMenu.vue';
 
-export type MenuItemAttrsProps = DefineAttrsProps<MenuItemProps, ListItemAttrsProps>;
-export type ListItemInstance = InstanceType<typeof UiListItem>
+export type MenuItemAttrsProps = DefineAttrsProps<MenuItemProps>;
+
+defineOptions({ inheritAttrs: false });
 
-const props = withDefaults(defineProps<MenuItemProps>(), {
-  icon: 'present',
-  suffixVisible: 'default',
-  suffixAttrs: () => ({ class: 'ui-button--text ui-menu-item__suffix' }),
-  listItemAttrs: () => ({ class: 'ui-menu-item' }),
+const attrs:ListItemAttrsProps = useAttrs();
+const filteredAttrs = computed(() => {
```

Comment:
Better name would called `attrsForListItem` or so, otherwise there is no indication what it's filtering and why.